zap/tests/irgen/mod.rs

```diff
@@ -354,3 +354,35 @@ async fn test_tagged_enum_quoted() {
 
 	runtime.run("Zap", output).await.unwrap();
 }
+
```

Comment:
Can we add some more complex tests like what you've been testing with locally?
As an example:
```
event Test = {
    from: Client,
    type: Reliable,
    call: SingleSync,
    data: (string | unknown | buffer | u16 | struct { bar: u8 } | enum { foo, bar, baz } | Instance (Part) | enum "test" {
        a { 
            b: u8
        },
        c { 
            d: u8
        }
    })?
}
```